contribs/gmf/src/controllers/abstractdesktop.js

```diff
@@ -54,34 +59,46 @@ gmf.AbstractDesktopController = function(
     labelKey: 'label',
     groupsKey: 'layer_name',
     groupValues: ['osm'],
-    projection: 'EPSG:21781',
+    projection: 'EPSG:' + (config.srid || 21781),
     url: fulltextsearchUrl
   }];
 
+  var viewConfig = {
+    projection: ol.proj.get('epsg:' + (config.srid || 21781))
+  };
+  goog.object.extend(viewConfig, config.mapViewConfig || {});
+
+  // TODO: Remove when we add the base layer switcher
+  var layer = new ol.layer.Tile();
   /**
    * @type {ol.Map}
    * @export
    */
   this.map = new ol.Map({
-    layers: [
-      new ol.layer.Tile({
-        source: new ol.source.OSM()
-      })
-    ],
-    view: new ol.View({
-      center: [0, 0],
-      zoom: 2
-    }),
-    controls: [
+    layers: [layer],
+    view: new ol.View(viewConfig),
+    controls: config.mapControls || [
       new ol.control.ScaleLine(),
       new ol.control.Zoom()
     ],
-    interactions: ol.interaction.defaults({
+    interactions: config.mapInteractions || ol.interaction.defaults({
       pinchRotate: false,
       altShiftDragRotate: false
     })
   });
 
+  // TODO: Remove when we add the base layer switcher
+  var parser = new ol.format.WMTSCapabilities();
```

Comment:
Just for information (because that's a "remove-me code block"), you can do the same with https://github.com/camptocamp/ngeo/blob/master/src/services/layerHelper.js#L107